apps/api/src/app/workflows-v2/usecases/build-variable-schema/build-available-variable-schema.usecase.ts

```diff
@@ -38,17 +39,17 @@ export class BuildAvailableVariableSchemaUsecase {
           additionalProperties: false,
         },
         steps: buildPreviousStepsSchema(previousSteps, workflow.payloadSchema),
-        payload: safePayloadSchema(workflow) || { type: 'object', description: 'Payload for the current step' },
+        payload: safePayloadSchema(workflow),
       },
       additionalProperties: false,
     } as const satisfies JSONSchemaDto;
   }
 }
-function safePayloadSchema(workflow: NotificationTemplateEntity): JSONSchemaDto | undefined {
+function safePayloadSchema(workflow: NotificationTemplateEntity): JSONSchemaDto {
```

Comment:
We already do this in `parsePayloadSchema`. If we want to always return an object, we can tweak the error handling there.